natlas-server/app/filters.py

```diff
@@ -11,6 +11,8 @@ def ctime(s, human=False):
     return parser.parse(s).strftime("%Y-%m-%d %H:%M")
 
 
-@bp.app_template_filter("hashpath")
-def hashpath(inhash):
-    return f"{inhash[0:2]}/{inhash[2:4]}/{inhash}.png"
+@bp.app_template_filter("get_screenshot_path")
+def get_screenshot_path(inhash: str, service: str = "HTTP"):
+    ext_map = {"HTTP": ".png", "HTTPS": ".png", "VNC": ".jpg"}
+
+    return f"{inhash[0:2]}/{inhash[2:4]}/{inhash}{ext_map[service]}"
```

Comment:
I understand this is a short-term fix since you mentioned you were going to add a better strategy, but we shouldn't make assumptions like this in the rendering logic just in case we change formats.

Reply:
Agreed, this just brings the rendering logic in line with the logic in screenshot processing:

https://github.com/natlas/natlas/blob/b715caa92c366ffe716a69d079bb58cb6a79b496/natlas-server/app/api/processing/screenshot.py#L29-L33

With the agent rearchitecture we'll probably lose backwards compatibility and it's the perfect time to make big changes to the data model too for a 0.7.0 release.